### PR TITLE
OHARA-1453 Add an loading indicator for pipeline new/edit page if the async requests are still pending

### PR DESCRIPTION
**Pull request by konekoya**
Work in progress. In this PR we're going to:

- Use React's new context API to share data that are needed for different components
- Refactor Header and pipeline modals so they can use context API
- Rewrite tests with context API

Todos:

- [x] Fix unit tests
- [ ] Use context API for components that consume cluster API

### CODE REVIEW

File: ohara-manager/client/src/components/context/ClusterContext.js [hunk 1]
```diff
@@ -22,10 +22,8 @@ class ClusterProvider extends React.Component {
 
     if (!_.isNull(clusterInfo)) {
       const { versionInfo } = clusterInfo;
-      this.setState({ versionInfo });
+      this.setState({ versionInfo, isLoading: false });
```
Comment (eechih) [inline ohara-manager/client/src/components/context/ClusterContext.js]: 因為 Http Request 已經結束，不管是否有拿到 clusterInfo，似乎都要將 isLoading 設成 false。

Comment (konekoya) [reply ohara-manager/client/src/components/context/ClusterContext.js]: Thanks. Haven't finished the work yet :-)

**Comment by konekoya** [comment]
This PR won't be included in v0.2 release. Therefore, removing the milestone

**Comment by konekoya** [comment]
We will create another PR in v03 to address this issue :)

**Pull request closed without merging**